model/src/main/kotlin/com/testerum/model/resources/http/response/xml_body/XmlRootJsObject.kt

```diff
@@ -0,0 +1,94 @@
+package com.testerum.model.resources.http.response.xml_body
+
+import jdk.nashorn.api.scripting.AbstractJSObject
+import org.jdom2.Attribute
+import org.jdom2.Content
+import org.jdom2.Document
+import org.jdom2.Element
+import org.jdom2.Text
+import org.jdom2.input.SAXBuilder
+import org.jdom2.xpath.XPathFactory
+
+const val XPATH_FUNCTION_NAME = "xpath"
+
+class XmlRootJsObject(private val xmlAsUtf8String: String) : AbstractJSObject() {
+    var rootElement: Element? = null
+    var document: Document? = null
```

Comment:
I like that we parse the XML only when needed.

We can write this nicer, without all the NPEs waiting to happen:
* make ``document`` a [lazy property](https://kotlinlang.org/docs/reference/delegated-properties.html#lazy).
* make ``rootElement`` a calculated property:
```kotlin
private var rootElement: Element
    get() = document.rootElement
```
* make both properties private and non-nullable